src/python/pants/backend/docker/goals/publish.py

```diff
@@ -74,6 +86,30 @@ async def push_docker_images(
             ]
         )
 
+    field_set = cast(PublishDockerImageFieldSet, request.field_set)
+    if options.use_buildx and field_set.platforms.value:
+        targets = await Get(Targets, Addresses([field_set.address]))
+        target = targets.expect_single()
+
+        build_setup = await Get(
+            DockerBuildSetup,
+            DockerBuildSetupRequest(
+                field_set=DockerPackageFieldSet.create(target),
```

Comment:
The publish goal is awkward for these targets because while they have a package already, its not usable. We need to re-invoke `docker buildx build` with the `--push` flag and `--platform`